src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -620,7 +620,8 @@ public void stopJenkins() throws Throwable {
         if (!proc.waitFor(60, TimeUnit.SECONDS) ) {
             System.err.println("Jenkins failed to stop within 60 seconds, attempting to kill the Jenkins process");
             proc.destroyForcibly();
-            throw new AssertionError("Jenkins failed to terminate within 60 seconds");
+            proc = null;
```

Comment:
What happens when test flakes and fails:

1. test runs well and going to [stopJenkins](https://github.com/jenkinsci/jenkins-test-harness/blob/6a1fdad76d6fa4998a5211f62008f234ea04bf87/src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java#L442) at the end.
2. stopJenkins [hit timeout](https://github.com/jenkinsci/jenkins-test-harness/blob/6a1fdad76d6fa4998a5211f62008f234ea04bf87/src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java#L620).
3. AssertExceptions thown as a result
4. [finally block](https://github.com/jenkinsci/jenkins-test-harness/blob/6a1fdad76d6fa4998a5211f62008f234ea04bf87/src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java#L389-L396) executed and because proc != null tries to [stop Jenkins again](https://github.com/jenkinsci/jenkins-test-harness/blob/6a1fdad76d6fa4998a5211f62008f234ea04bf87/src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java#L390)
5. ConnectionException throws [here](https://github.com/jenkinsci/jenkins-test-harness/blob/6a1fdad76d6fa4998a5211f62008f234ea04bf87/src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java#L619)

Reply:
My suggestion is to set proc = null (like [here](https://github.com/jenkinsci/jenkins-test-harness/blob/6a1fdad76d6fa4998a5211f62008f234ea04bf87/src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java#L594-L595)) instead of throwing AssertError